.github/workflows/yocto-build-deploy.yml

```diff
@@ -0,0 +1,1174 @@
+        description: balena API key for deploying releases to staging # TODO: Different for staging and prod - add another Secret for staging key
+        required: false
+      # Dockerhub secrets are used only for pulling the helper image for "Prepare files for S3" step - if we simplify this to not use the 
+      # helper image, these secrets can be removed
+      DOCKERHUB_USER:
+        description: Dockerhub user for pulling private helper images
+        required: false
+      DOCKERHUB_TOKEN:
+        description: Dockerhub token for pulling private helper images
+        required: false
+      SIGN_KMOD_KEY_APPEND:
+        description: Base64-encoded public key of a kernel module signing keypair
+        required: false
+      # TODO: can this be the same as BALENA_API_KEY?
+      SIGN_API_KEY:
+        description: balena API key that provides access to the signing server
+        required: false
+    inputs:
+      runs-on:
+        description: The runner labels to use for the job(s)
+        required: false
+        type: string
+        default: >
+          [
+            "self-hosted",
+            "X64"
+          ]
+      device-repo:
+        description: balenaOS device repository (owner/repo)
+        required: false
+        type: string
+        default: ${{ github.repository }}
+      device-repo-ref:
+        description: balenaOS device repository tag, branch, or commit to build
+        required: false
+        type: string
+        default: ${{ github.ref }}
+      meta-balena-ref:
+        description: meta-balena ref if not the currently pinned version
+        required: false
+        type: string
+      yocto-scripts-ref:
+        description: balena-yocto-scripts ref if not the currently pinned version
+        required: false
+        type: string
+      machine:
+        description: yocto board name
+        required: true
+        type: string
+      environment:
+        description: The GitHub Environment to use for the job(s) (production, staging, etc.)
+        required: false
+        type: string
+        default: production 
+      test-environment:
+        description: The BalenaCloud environment you want tests to target. Available options - production, staging, balenaos-balenamachine
+        required: false
+        type: string
+        default: balenaos-balenamachine
+      deploy-s3:
+        description: Whether to deploy images to S3
+        required: false
+        type: boolean
+        default: ${{ github.event_name == 'push' }} # Deploy to s3 only on creation of new tags - a "push " event. We're happy with just push events, as the caller workflow only starts this on pushed tags, matched with semver
+      deploy-hostapp:
+        description: Whether to deploy a hostApp container image to a balena environment
+        required: false
+        type: boolean
+        default: true # we always want to deploy the hostapp - as eventually we'll use it for HUP tests
```

Comment:
If we always want to deploy them, why is this an option? :-)

Also: Maybe worth rephrasing to express the correct interpretation of "eventually we'll use it for HUP tests". Either of:

1. Some day in the future we'll change the HUP tests to use the hostapp
2. Some step further down the workflow will use the hostapp

Reply:
Mainly the option was there in case we want to do a manual run of the build and/test - and you'll save a (admittedly small) bit of time. But I see the point!

Agreed on the rewording